base_layer/mmr/src/sparse_merkle_tree/tree.rs

```diff
@@ -271,9 +271,10 @@ impl<H: Digest<OutputSize = U32>> SparseMerkleTree<H> {
         Ok(node.map(|n| n.as_leaf().unwrap().value()))
     }
 
-    /// Constructs a Merkle proof for the value at location `key`.
-    pub fn build_proof(&self, key: &NodeKey) -> Result<MerkleProof<H>, SMTError> {
-        let mut path = Vec::new();
+    /// Construct the data structures needed to generate the merkle proofs. Although this function returns a struct
+    /// of type `ExclusionProof` it is not really a valid (exclusion) proof. The constructors do additional
+    /// validation before passing the structure on. For this reason, this method is `private` outside of the module.
+    pub(crate) fn non_failing_exclusion_proof(&self, key: &NodeKey) -> Result<ExclusionProof<H>, SMTError> {
```

Comment:
Returning a wrapped `ExclusionProof` here feels wrong, even though any other approach (using a type alias for clarity, defining a custom return struct, or returning a wrapped tuple with the sibling and leaf data) seems like overkill. Maybe just rename the function to something like `get_siblings_and_leaf_for_proofs` as an added indicator that you're not really getting a proof out? But this is more of a nit.

Reply:
or `build_pre_proof`? `build_proof_candidate`?

Reply:
Even better! I think I like `build_proof_candidate` best. Seems like a good compromise between clarity and creating internal types unnecessarily.